docs/ordinals/nft.md

```diff
@@ -0,0 +1,162 @@
+---
+sidebar_position: 2
+---
+
+# Non Funglible Tokens - NFTs
+
+
+To create a smart contract that will carry an NFT, have your smart contract extend the `OneSatNFT` class:
+
+```ts
+import { method, prop, assert, ByteString, sha256, Sha256 } from "scrypt-ts";
+import { OneSatNFT } from "scrypt-ord";
+
+export class HashPuzzleNFT extends OneSatNFT {
+  @prop()
+  hash: Sha256;
+
+  constructor(hash: Sha256) {
+    super();
+    // Important: Call `init` after the `super()` statement.
+    this.init(...arguments);
+    this.hash = hash;
+  }
+
+  @method()
+  public unlock(message: ByteString) {
+    assert(this.hash === sha256(message), "hashes are not equal");
+  }
+}
+```
+
+The contract above represents an NFT that can be unlocked / transferred by providing the secret pre-image of a hash value.
+
+## Inscribe
+
+The following code demonstrates how deploy / inscribe the NFT contract:
+
+```ts
+HashPuzzleNFT.loadArtifact();
+
+const text = "Hello sCrypt and 1Sat Ordinals";
+
+const message = toByteString('secret string', true);
+const hash = sha256(message);
+
+const instance = new HashPuzzleNFT(hash);
+
+const signer = getDefaultSigner();
+await instance.connect(signer);
+
+const inscriptionTx = await instance.inscribeTextNft(text);
+console.log("Inscribed NFT: ", inscriptionTx.id);
+```
+
+The `inscribeTextNft` first inscribes the locking script with the specified text and then deploys the contract.
+
+Among text the inscription can contain many other types of data. Here's how you can conveniently inscribe an image:
+
+```ts
+// ...
+
+const bb = readFileSync(join(__dirname, "..", "..", "logo.png")).toString("base64");
+
+const tx = await instance.inscribeImageNft(bb, ContentType.PNG);
+console.log("Inscribed NFT: ", tx.id);
+```
+
+In fact the data type can be arbitrary. It only depends on the Ordinals wallet you're using to support that data type.
+
+```ts
+const tx = await instance.inscribe({
+  content: `your content in hex`,
+  contentType: `your contentType`,
+});
+console.log("Inscribed NFT: ", tx.id);
+```
+
+The value `contentType` must be a MIME-type string. The [`ContentType`](https://github.com/sCrypt-Inc/scrypt-ord/blob/master/src/contentType.ts) object contains common MIME-types.
+
+## Transfer
+
+You can easily transfer a deployed NFT to an Ordinals address by passing a `transfer` value via the method call parameters.
+
+```ts
+// ... deploy code from above
+
+const { tx: transferTx } = await instance.methods.unlock(
+    message, 
+    {
+        transfer: new OneSatNFTP2PKH(
```

Comment:
Explain OneSatNFTP2PKH. Put it in a separate built-in, like https://docs.scrypt.io/how-to-write-a-contract/built-ins#standard-libraries

Reply:
Explain  added